### PR TITLE
Enable chunked proving

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR builds on #1032, removing the checks that disabled chunked proving, and adding a test.

Only one of the added tests is enabled; the others are commented out to avoid increasing CI time too much.

Stats from my (admittedly slow) machine:
```
$ cargo test --release 2_to_16_unchunked_generic_gate_pub -- --nocapture

running 1 test
- time to create prover index: 39s
test tests::chunked::test_2_to_16_unchunked_generic_gate_pub has been running for over 60 seconds
- time to create proof: 18s
- time to verify: 337ms
test tests::chunked::test_2_to_16_unchunked_generic_gate_pub ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 191 filtered out; finished in 63.40s

$ cargo test --release 2_to_17_chunked_generic_gate_pub -- --nocapture

running 1 test
test tests::chunked::test_2_to_17_chunked_generic_gate_pub has been running for over 60 seconds
- time to create prover index: 154s
- time to create proof: 45s
- time to verify: 351ms
test tests::chunked::test_2_to_17_chunked_generic_gate_pub ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 191 filtered out; finished in 213.64s

$ cargo test --release 2_to_18_chunked_generic_gate_pub -- --nocapture

running 1 test
test tests::chunked::test_2_to_18_chunked_generic_gate_pub has been running for over 60 seconds
- time to create prover index: 623s
- time to create proof: 129s
- time to verify: 395ms
test tests::chunked::test_2_to_18_chunked_generic_gate_pub ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 191 filtered out; finished in 801.81s
```
Roughly, for each doubling of the number of constraints:
* The prover index appears to take ~4x as long to generate (including the SRS and domain)
* The proving time appears to take ~2.5x as long
* The verification time increased about 8%

### CODE REVIEW

**Review by querolita — APPROVED** [review]
Interesting numbers, super helpful having tests to benchmark it.

File: kimchi/src/tests/chunked.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    test_generic_gate_with_srs_override(20, Some(1 << 16))
+}*/
+
+// Disabled, too slow
+/*#[test]
+fn test_2_to_18_chunked_generic_gate_pub() {
+    test_generic_gate_with_srs_override(18, Some(1 << 16))
+}*/
+
+#[test]
```
Comment (jspada) [inline kimchi/src/tests/chunked.rs:106]: nice

File: poly-commitment/src/evaluation_proof.rs [hunk 2]
```diff
@@ -117,8 +117,6 @@ impl<G: CommitmentCurve> SRS<G> {
             let mut plnm_evals_part = {
                 // For now just check that all the evaluation polynomials are the same degree so that we
                 // can do just a single FFT.
-                // Furthermore we check they have size less than the SRS size so we don't have to do chunking.
-                // If/when we change this, we can add more complicated code to handle different degrees.
```
Comment (jspada) [inline poly-commitment/src/evaluation_proof.rs:121]: I guess there was not complicated code to add here as part of this PR. :-)

File: kimchi/src/tests/chunked.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+fn test_2_to_17_chunked_generic_gate_pub() {
+    test_generic_gate_with_srs_override(17, Some(1 << 16))
+}
```
Comment (jspada) [inline kimchi/src/tests/chunked.rs:109]: May be good to have a negative test case if you can think of any.  Maybe fiddling with public input and seeing that it doesn't verify.

File: kimchi/src/tests/chunked.rs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
```
Comment (jspada) [inline kimchi/src/tests/chunked.rs:110]: Since chunking impacts permutation argument, maybe be good to have a test case with some non-trivial wiring.

**Review by jspada — APPROVED** [review]
Approved with some suggestions.

**Pull request merged**